code/cryptography/src/autokey_cipher/autokey.c

```diff
@@ -0,0 +1,87 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+void flush_stdin()
+{
+    while(fgetc(stdin) != '\n');
+}
+
+void AutoKeyCipher(char** Text, char key, int encrypt)
+{
+    int textSize, i, nextKey, keyValue, result;
```

Comment:
Declare these near where they are first used; no need to declare everything at the top of the function.